scripts/speedtest.py

```diff
@@ -23,23 +24,31 @@ def runSpeedtest():
 
         a = os.popen(speedtestCommand).read()
         print('ran')
-        #split the 3 line result (ping,down,up)
-        lines = a.split('\n')
-        print(a)
+
         ts = time.time()
         date =datetime.datetime.fromtimestamp(ts).strftime('%d.%m.%Y %H:%M:%S')
         print(date)
         #if speedtest could not connect set the speeds to 0
         if "Cannot" in a:
-                p = 100
+                p = 0
                 d = 0
                 u = 0
+		isp = ''
+		lat=0
+		lon=0	
         #extract the values for ping down and up values
         else:
-                p = lines[0][6:11]
-                d = lines[1][10:14]
-                u = lines[2][8:12]
-        print(date,p, d, u)
+	        #Parse json output, load as dictionary                          
+        	results_dict = json.loads(a)                                    
+
+                p = round( results_dict['ping'] ,2)
+                d = round( results_dict['download'] /1000.0 / 1000.0, 2)
+                u = round( results_dict['upload'] /1000.0 / 1000.0, 2)
+		isp = results_dict['client']['isp']
+		lat = results_dict['client']['lat'] 
+		lon = results_dict['client']['lon']
+
```

Comment:
If we take advantage of the get() method on dictionaries and set the default value if the key is not found, we can simplify this some, and make it more robust.

```python
try:
    results = json.loads(a)
except json.decoder.JSONDecodeError:
    results = {}

p = results.get('ping', 0)
d = round(results.get('download', 0)/1000000.0, 2)
u = round(results.get('upload', 0)/1000000.0, 2)
isp = results.get('client', {}).get('isp', '')
lat = results.get('client', {}).get('lat', 0)
lon = results.get('client', {}).get('lon', 0)
```

Reply:
I verified this works with the failure case as well.